platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/state/VirtualMerkleStateInitializer.java

```diff
@@ -67,110 +77,166 @@ private VirtualMerkleStateInitializer() {}
+                        createAccountsVM(useMerkleDb, storageDir, totalAccounts);
+                logger.info(LOGM_DEMO_INFO, "accounts VM = {}, DS = {}", virtualMap, virtualMap.getDataSource());
                 virtualMap.registerMetrics(platform.getContext().getMetrics());
                 state.setVirtualMap(virtualMap);
             }
 
-            final long maximumNumberOfKeyValuePairsCreation =
-                    virtualMerkleConfig.getMaximumNumberOfKeyValuePairsCreation();
-            if (state.getVirtualMapForSmartContracts() == null && maximumNumberOfKeyValuePairsCreation > 0) {
+            final long maximumNumberOfKeyValuePairs = virtualMerkleConfig.getMaximumNumberOfKeyValuePairsCreation();
+            logger.info(LOGM_DEMO_INFO, "max KV pairs = {}", maximumNumberOfKeyValuePairs);
+            if (state.getVirtualMapForSmartContracts() == null && maximumNumberOfKeyValuePairs > 0) {
                 logger.info(
                         LOGM_DEMO_INFO,
                         "Creating virtualmap for max {} key value pairs.",
-                        maximumNumberOfKeyValuePairsCreation);
-                final SmartContractMapKeySerializer keySerializer = new SmartContractMapKeySerializer();
-                final VirtualLeafRecordSerializer<SmartContractMapKey, SmartContractMapValue> leafRecordSerializer =
-                        new VirtualLeafRecordSerializer<>(
-                                (short) 1,
-                                keySerializer.getSerializedSize(),
-                                new SmartContractMapKeyBuilder(),
-                                (short) 1,
-                                SmartContractMapValue.getSizeInBytes(),
-                                new SmartContractMapValueBuilder(),
-                                false);
-
-                final JasperDbBuilder<SmartContractMapKey, SmartContractMapValue> jasperDbBuilder = new JasperDbBuilder<
-                                SmartContractMapKey, SmartContractMapValue>()
-                        .virtualLeafRecordSerializer(leafRecordSerializer)
-                        .virtualInternalRecordSerializer(new VirtualHashRecordSerializer())
-                        .keySerializer(keySerializer)
-                        .storageDir(pathToJasperDBStorageDir)
-                        .maxNumOfKeys(maximumNumberOfKeyValuePairsCreation)
-                        .hashesRamToDiskThreshold(0)
-                        .preferDiskBasedIndexes(false);
-
+                        maximumNumberOfKeyValuePairs);
                 final VirtualMap<SmartContractMapKey, SmartContractMapValue> virtualMap =
-                        new VirtualMap<>("smartContracts", jasperDbBuilder);
+                        createSmartContractsVM(useMerkleDb, storageDir, maximumNumberOfKeyValuePairs);
+                logger.info(LOGM_DEMO_INFO, "SC VM = {}, DS = {}", virtualMap, virtualMap.getDataSource());
```

Comment:
Quite a few of these info logs probably should be debug.  Info is actually a fairly high logging level.